python/pacemaker/_cts/environment.py

```diff
@@ -192,19 +193,22 @@ def _detect_systemd(self):
 
     def _detect_syslog(self):
         """Detect the syslog variant in use on the target node."""
-        if "syslogd" not in self.data:
-            if self["have_systemd"]:
-                # Systemd
-                (_, lines) = self._rsh(self._target, r"systemctl list-units | grep syslog.*\.service.*active.*running | sed 's:.service.*::'", verbose=1)
-                self["syslogd"] = lines[0].strip()
-            else:
-                # SYS-V
-                (_, lines) = self._rsh(self._target, "chkconfig --list | grep syslog.*on | awk '{print $1}' | head -n 1", verbose=1)
-                self["syslogd"] = lines[0].strip()
+        if "syslogd" in self.data:
+            return
+
+        if self["have_systemd"]:
+            # Systemd
+            (_, lines) = self._rsh(self._target, r"systemctl list-units | grep syslog.*\.service.*active.*running | sed 's:.service.*::'", verbose=1)
+        else:
+            # SYS-V
+            (_, lines) = self._rsh(self._target, "chkconfig --list | grep syslog.*on | awk '{print $1}' | head -n 1", verbose=1)
+
+        with suppress(IndexError):
+            self["syslogd"] = lines[0].strip()
 
-            if "syslogd" not in self.data or not self["syslogd"]:
-                # default
-                self["syslogd"] = "rsyslog"
+        if "syslogd" not in self.data or not self["syslogd"]:
+            # default
+            self["syslogd"] = "rsyslog"
```

Comment:
I'd rather not set this if not found, instead just guard the use in audits.py

Reply:
And also the use in cts-lab.